File: NapakalakiRuby/lib/monster.rb
# encoding: utf-8
# 
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Monster
      
  public
      
  # Constructor
  def initialize(name, combat_level, bc, prize, level_change_against_cultist_player)
    @name = name
    @combatLevel = combat_level
    @badConsequence = bc
    @prize = prize
    @levelChangeAgainstCultistPlayer = level_change_against_cultist_player
  end
  
  def Monster.new_basic_monster(name, combat_level, bc, prize)
    new(name, combat_level, bc, prize, 0)
  end
  
  #Getters
  def get_name
    @name
  end
  
  def get_combat_level
    @combatLevel
  end
  
  def get_bad_consequence
    @badConsequence
  end
  
  def get_levels_gained
    @prize.get_levels()
  end
  
  def get_treasures_gained
    @prize.get_treasures()
  end
  
  def get_prize
    @prize
  end
  
  def get_level_change_against_cultist_player
    @levelChangeAgainstCultistPlayer
  end
  
  def get_combat_level_against_cultist_player
    get_combat_level+get_level_change_against_cultist_player
  end
    
  # Devuelve un string con los datos del monstruo
  def to_s
    
    "Monster: #{@name}; CombatLevel: #{@combatLevel}; \n\t#{@badConsequence.to_s}; " +
      "\t#{@prize.to_s}"
    
  end
  
end

File: Napakalaki/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author caracuel
 */
public class Monster {
    
    // Variables
    private String name;
    private int combatLevel;
    
    private BadConsequence bc;
    private Prize prize;
    
    private int levelChangeAgainstCultistPlayer;
        
    // Constructores
    Monster(String name, int combatLevel, BadConsequence bc, Prize prize){
        
        setName(name);
        setCombatLevel(combatLevel);
        setBC(bc);
        setPrize(prize);
        setLevelChangeAgainstCultistPlayer(0);
        
    }
    
    Monster(String name, int combatLevel, BadConsequence bc, Prize prize,
            int levelChangeAgainstCultistPlayer){
        
        setName(name);
        setCombatLevel(combatLevel);
        setBC(bc);
        setPrize(prize);
        setLevelChangeAgainstCultistPlayer(levelChangeAgainstCultistPlayer);
        
    }
    
    // Getters
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return bc;
    }
    
    public Prize getPrize(){
        return prize;
    }
    
    ///////////////////////////////////////////////////////////////////////////
    // Si se devuelve prize no son necesarios los dos siguientes métodos,
    // pero en el esquema UML que tenemos aparecen
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    ///////////////////////////////////////////////////////////////////////////
    
    public int getLevelChangeAgainstCultistPlayer(){
        return levelChangeAgainstCultistPlayer;
    }
        
    // Setters
    // No me gusta que sean public...
    public void setName(String name){
        this.name = name;
    }
    
    public void setCombatLevel(int combatLevel){
        this.combatLevel = combatLevel;
    }
    
    public void setBC(BadConsequence bc){
        this.bc = bc;
    }
    
    public void setPrize(Prize prize){
        this.prize = prize;
    }
    
    public void setLevelChangeAgainstCultistPlayer(int lccp){
        this.levelChangeAgainstCultistPlayer = lccp;
    }
    
    // Métodos incluidos en la sesión 4
    public int getCombatLevelAgainstCultistPlayer(){
        return getCombatLevel()+getLevelChangeAgainstCultistPlayer();
    }
    
    // Utils
    @Override
    public String toString(){
        return "Name = " + name +
                ", CombatLevel = " + Integer.toString(combatLevel) +
                ", BadConsequence = { " + bc.toString() +
                " }, Prize = {" + prize.toString() + " }";
    }
    
}